demos-europe/demosplan-core
# feat(refs T35963): enable split statements

**Ticket:** https://yaits.demos-deutschland.de/T35963

We want to have the feature of splitting statements in the core, too, but without the pipeline. 

- Moved all necessary file from addon to core (all but the processing page)
- add `splitStatementView` directly in twig instead of `addonWrapper`
- `addonWrapper` (hook: split.statement.ai) is now in `SplitStatementView`
- `addonWrapper` now emits an event with boolean payload if it has loaded successfully or not
- `addonWrapper` forwards `segementationStatus:change` event from addon component
- changes in `SplitStatementView`:
1. Listen to loaded event from addon. If not loaded (no addon installed) `fetchSegmentDraftList` and `fetchInitialData`. If there is no segmentation yet and no segmentDraftList `setInitialData` for SegmentDraftList and `fetchTags`. 
2. Listen to `segmentationStatus:change` event forwared from addon and `setSegmentationStatus`
3. Remove `fetchSegmentDraftList` and `fetchInitialData` from mounted(), because we need to distinguish if addon is installed or not. Fetching segments with ai is handled in Addon. 
- In store add action for initializing data without segmentDraftList form backend and an own action for fetchTags. 

**Open questions:**

- We need to listen to event of the `SplitStatementAi`. Therefor the `AddonWrapper` needs to forward the event. I'm not sure about this solution, because the `AddonWrapper` is a generic component for all addons and it feels weird to forward an event from a specific component. Any ideas? My only other idea was to use vue bus.
- To check if an addon was loaded or not (aka is installed or not) I added an event `isLoaded` to the `AddonWrapper` which is always emitted and has a boolean as payload (true = addon has loaded, false = addon has not loaded). Is there any other way to check if an addon is installed or do we want to do it like this?
- I moved the logic for getting data from the pipeline to the addon in `SplitStatementAi` since we can access a core store from the addon. The component emits an event for changing the segmentationStatus. We can use this for the click event, too, so we don't need to forward two different events. The only pipeline logic in core is now in `SplitStatementStore` (like before) and some prosemirror stuff (in `/lib/prosemirror`). Do we leave it like this? I'm not sure how we can separate this. 


### How to review/test
- Install addon with correct branch
- You should be able to split statements with addon enabled or disabled
- Addon enabled: Should work life before
- Addon disabled: You should get to the split statement view immediately without the processing page. There everything should work as usual.
